### DISCUSSION TITLE
Unable to install METV11.1.0 and METplus V5.1.0 containers

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2376] [Installation] [answered]
I am trying to install the MET container on the HPC and am using a singularity command as follows:
singularity pull docker://dtcenter/met:11.1.0
The error I receive is:
FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://dtcenter/met:11.1.0: Requesting bear token: invalid status code from registry 503 (Service Unavailable)

I checked the tags at: https://github.com/dtcenter/MET/tags, since this error is similar to the one I had when installing the METplus container. The tag for V11.1.0 seems identical to the one I specified in the singularity command. I don't understand why I'm getting the error or what to do about it. Do you have a resolution? Thanks.

Ditto the above problem when attempting to install the METplus container. The tag info seems to match what I'm putting in the command: singularity pull docker://dtcenter/metplus:5.1.0. I am getting a similar error:
FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://dtcenter/metplus:5.1.0: error pinging docker registry registry-1.docker.io: Get https://registry-1.docker.io/v2/: EOF

**Comment by JohnHalleyGotway** [comment]
@jwraby I'll say that I have not tried to directly replicate the problem you're having. However, I did do some Googling on it and found the following.

On [this page](https://awesome-workshop.github.io/docker-cms/05-singularity/index.html), search for `failed to get checksum`. They claim that this is a singularity bug that can be fixed by deleting the `~/.docker/config.json` file.

Please try that out and let me know how it goes.

**Reply by jwraby** [reply, marked as answer]
Hi John - 

Thanks for your suggestion. I looked for that file on the HPC but could not find it. I re-ran the same commands and successfully installed both containers. 
singularity pull docker://dtcenter/met:11.1.0
singularity pull docker://dtcenter/metplus:5.1.0

**Reply by JohnHalleyGotway** [reply]
Great, glad you were able to move past the issue. Its always a little unsettling when problems resolve themselves without an obvious explanation, but I'm glad to hear its working.

I'll note that singularity was renamed to [Apptainer](https://apptainer.org/news/community-announcement-20211130/) a while back. Just a name change for you to be aware of.

I'll go ahead and mark this discussion as being answered.

**Comment by jwraby** [comment]
I agree with you on the uncertainty about why the issue had errors before
and not now. I was wondering if maybe the repo was having an issue of some
kind.  Thanks for the update on Apptainer. I had noticed that the
Singularity prompt changed to Apptainer recently, but I wasn't sure if
Singularity was replaced by Apptainer or whether there was a name change.
Now I know. Concur on this discussion being answered.

 

Thanks.

 

R/

John
